optuna/integration/allennlp.py

```diff
@@ -9,17 +10,127 @@
 import optuna
 from optuna._experimental import experimental
 from optuna._imports import try_import
+from optuna import load_study
+from optuna import Trial
+
 
 with try_import() as _imports:
     import allennlp
     import allennlp.commands
     import allennlp.common.util
-    from allennlp.training import EpochCallback
 
+# EpochCallback is conditionally imported for CI because allennlp is not available on
+# the eivironment that builds documentation.
 if _imports.is_successful():
     import _jsonnet
+    from allennlp.training import EpochCallback
 else:
-    EpochCallback = object  # NOQA
+    # `allennlp.training.EpochCallback` is a subclass of `Registerable`
+    # https://docs.allennlp.org/master/api/training/trainer/#epochcallback
+    # On the other hand, `EpochCallback` defined here is not `Registerable`,
+    # which causes a mypy checking feailure.
+    class EpochCallback:  # type: ignore
+        """Stub for EpochCallback.
+
+        This class is introduced for documentation CI.
+
+        """
+
+        @classmethod
+        def register(cls: Any, *args: Any, **kwargs: Any) -> Callable:
+            def wrapper(subclass: Any, *args: Any, **kwargs: Any) -> None:
+                return subclass
+
+            return wrapper
+
+
+PPID = os.getppid()
+
+
+def _create_pruner() -> Optional[optuna.pruners.BasePruner]:
+    pruner_class = os.getenv("{}_OPTUNA_ALLENNLP_PRUNER_CLASS".format(PPID))
+    if pruner_class is None:
+        return None
+
+    pruner_params = _get_environment_variables_for_pruner()
+
+    pruner = None  # type: Any
+    if pruner_class == "HyperbandPruner":
+        pruner = optuna.pruners.HyperbandPruner
+    elif pruner_class == "MedianPruner":
+        pruner = optuna.pruners.MedianPruner
+    elif pruner_class == "PercentilePruner":
+        pruner = optuna.pruners.PercentilePruner
+    elif pruner_class == "ThresholdPruner":
+        pruner = optuna.pruners.ThresholdPruner
+    elif pruner_class == "SuccessiveHalvingPruner":
+        pruner = optuna.pruners.SuccessiveHalvingPruner
+    elif pruner_class == "NopPruner":
+        pruner = optuna.pruners.NopPruner
```

Comment:
Doesn't this work because this logic looks verbose to me?

```suggestion
    pruner = getattr(optuna.pruners, pruner_class, None)
```

Reply:
Amazing suggestion! Thank you.